### PR TITLE
Add adjugate_matrix

### LINKED ISSUE #1269: [Problem proposal] Adjugate matrix
Problem name: Adjugate matrix
(Optional) Problem ID: adjugate_matrix

## Problem

Given $N \times N$ matrix $A = \lbrace a_{ij} \rbrace$ with entries in $\mathbb{Z}/p\mathbb{Z}$, print $\text{adj } A = \lbrace (-1)^{i+j}M_{ji} \rbrace$, where $M_{ij}$ is the determinant of the matrix that is obtained from $A$ by removing its $i$-th row and $j$-th column. 

##  Constraint

- $1 \leq N \leq 500$

## Solution / Reference
- All solutions rely on $A \text{ adj }A = I_n \det A$. [Wiki article](https://en.wikipedia.org/wiki/Adjugate_matrix#Properties): 
  - If $A$ is invertible, then $\text{adj }A = A^{-1} \det A$;
  - If $\text{rank }A < n-1$, then $\text{adj }A = 0$;
  - If $\text{rank }A = n-1$, then $\text{adj }A = \alpha xy^\top$, where $Ax = 0$ and $A^\top y = 0$. Find $\alpha$ by finding any non-zero $M_{ij}$.
- Can also be found via Schur's complement by embedding $A$ into an invertible matrix or via Frobenius normal form.

## (Optional) Input

$N$
$a_{11}$ $a_{12}$ ... $a_{1N}$
$a_{21}$ $a_{22}$ ... $a_{2N}$
:
$a_{N1}$ $a_{N2}$ ... $a_{NN}$

## (Optional) Output

$b_{11}$ $b_{12}$ ... $b_{1N}$
$b_{21}$ $b_{22}$ ... $b_{2N}$
:
$b_{N1}$ $b_{N2}$ ... $b_{NN}$

$b_{ij} = (-1)^{i+j} M_{ji}$.

## (Optional) Note / Disucussion

- Was used in [Swiss Subregional Contest 2024](https://swiss.eolymp.space/en/s/results) (problem A), no solves though.
- Can reuse matrix inverse as a base, but need to add a few corner $\text{rank }A = n-1$ test cases.

### PR DESCRIPTION
**Pull request by adamant-pwn**
Resolves #1269.

I used matrix_inverse as a template, but also added the following generators:
- `almost_perm_max_random.cpp`: Same as `perm_max_random.cpp`, but changes a random item into `0`.
- `almost_identity.cpp`: An identity matrix, but random item is changed into `0`.
- `highrank_max_random.cpp`: A random matrix, but with 50% chance a random row is changed into a linear combination of two other rows, and with an independent 50% chance same happens to columns.

I think I need some help with Japanese strings for `task.md`.

### CODE REVIEW

**Comment by maspypy** [comment]
Here is the Japanese statement.
```
@{lang.en}
Given $N \times N$ matrix $A = \lbrace a_{ij} \rbrace$ with entries in $\mathbb{Z}/@{param.MOD}\mathbb{Z}$, print $\operatorname{adj} A = \lbrace (-1)^{i+j}M_{ji} \rbrace$, where $M_{ij}$ is the determinant of the matrix that is obtained from $A$ by removing its $i$-th row and $j$-th column. 
@{lang.ja}
$\mathbb{Z}/@{param.MOD}\mathbb{Z}$ 成分の $N\times N$ 行列 $A = \lbrace a_{ij} \rbrace$ が与えられます．
$A$ の余因子行列 $\operatorname{adj} A = \lbrace (-1)^{i+j}M_{ji} \rbrace$ を出力してください．ただし $M_{ij}$ は $A$ から $i$ 行目，$j$ 列目を取り除くことによって得られる行列の行列式です．
@{lang.end}
```

File: linear_algebra/adjugate_matrix/task.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+$a_{N1}$ $a_{N2}$ ... $a_{NN}$
+```
+
+## @{keyword.output}
+
+@{lang.en}
+Print $b_{ij} = (-1)^{i+j} M_{ji}$:
+@{lang.ja}
+
+@{lang.end}
```
Comment (adamant-pwn) [inline linear_algebra/adjugate_matrix/task.md:31]: I assume we also need a translation here?

Comment (maspypy) [reply linear_algebra/adjugate_matrix/task.md:31]: @{lang.en}
Print $b_{ij} = (-1)^{i+j} M_{ji}$:
@{lang.ja}
$b_{ij} = (-1)^{i+j} M_{ji}$ を出力してください :
@{lang.end}

Comment (adamant-pwn) [reply linear_algebra/adjugate_matrix/task.md:31]:
```suggestion
@{lang.en}
Print $b_{ij} = (-1)^{i+j} M_{ji}$:
@{lang.ja}
$b_{ij} = (-1)^{i+j} M_{ji}$ を出力してください :
@{lang.end}
```

**Comment by maspypy** [comment]
I implemented it as well, and I was able to confirm an AC!

How about adding a test case for `nontrivial_frobenius_form` for this problem as well? It can be added simply by copying it from `characteristic_polynomial`.

**Comment by maspypy** [comment]
I also suggest adding a test case for $n=1$. 
 I think it would be ok to simply add it as example_03.

**Review by maspypy — APPROVED** [review]

File: linear_algebra/adjugate_matrix/info.toml
Comment (maspypy) [inline linear_algebra/adjugate_matrix/info.toml:1]: I also checked with @yosupo06 , but it seems that even if you write CXXFLAGS in info.toml, it won't be added to any compilation options, making it meaningless

Comment (adamant-pwn) [reply linear_algebra/adjugate_matrix/info.toml:1]: Oops, I was checking it locally and forgot to remove :slightly_smiling_face: 

This is why I'm using a tester's solution as model here, and not my own

**Comment by adamant-pwn** [comment]
Thanks for the review! I have updated the PR.

File: linear_algebra/adjugate_matrix/task.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+$b_{N1}$ $b_{N2}$ ... $b_{NN}$
+```
+
+## @{keyword.sample}
+
+@{example.example_00}
+
+@{example.example_01}
+
+@{example.example_02}
```
Comment (maspypy) [inline linear_algebra/adjugate_matrix/task.md:46]:
```suggestion
## @{keyword.sample}

@{example.example_00}

@{example.example_01}

@{example.example_02}

@{example.example_03}
```

Comment (adamant-pwn) [reply linear_algebra/adjugate_matrix/task.md:46]: Thanks! Done :slightly_smiling_face:

**Review by maspypy — APPROVED** [review]

**Comment by maspypy** [comment]
Thank you! I will merge.

**Pull request merged**